test/e2e/main_test.go

```diff
@@ -12,39 +12,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package e2e_test
+package e2e
 
 import (
 	"fmt"
 	"os"
 	"path/filepath"
 	"runtime"
 	"testing"
-
-	e2e "github.com/tetratelabs/getenvoy/test/e2e/util"
 )
 
-// getEnvoy is the absolute path to the "getenvoy" binary used in all tests.
-var getEnvoy = e2e.GetEnvoy
-
 //nolint:golint
 const E2E_GETENVOY_BINARY = "E2E_GETENVOY_BINARY"
 
-// TestMain ensures the "getenvoy" binary is valid.
+// TestMain ensures the "getenvoy" binary is valid and we can read an Envoy reference (version) to run.
 func TestMain(m *testing.M) {
 	// As this is an e2e test, we execute all tests with a binary compiled earlier.
-	path, err := getEnvoyPath()
+	path, err := readGetEnvoyPath()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, `failed to start e2e tests due to an invalid "getenvoy" binary: %v`, err)
 		os.Exit(1)
 	}
-	e2e.GetEnvoyPath = path
+	getEnvoyPath = path
+
+	// Now, lookup the version we are to use
+	referenceTxt := "../../pkg/reference.txt"
```

Comment:
This is reading the file instead of using module embedding intentionally as it is cleaner than mixing the top-level module into this.